src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -77,6 +79,12 @@ internal class PullRequestBuilder : IPullRequestBuilder
     private readonly IBasicBarClient _barClient;
     private readonly ILogger<PullRequestBuilder> _logger;
 
+    private record DependencyCategories(
+    List<DependencyUpdateSummary> NewDependencies,
+    List<DependencyUpdateSummary> RemovedDependencies,
+    List<DependencyUpdateSummary> UpdatedDependencies
+    );
```

Comment:
```suggestion
    private record DependencyCategories(
        List<DependencyUpdateSummary> NewDependencies,
        List<DependencyUpdateSummary> RemovedDependencies,
        List<DependencyUpdateSummary> UpdatedDependencies);
```

Reply:
You can also replace `List<>` with `IReadOnlyCollection<>` if you want to express more immutability